### DISCUSSION TITLE
Caching

### DISCUSSION THREAD

**Discussion by ankita-sharma2019** [mockoon/mockoon#682] [Help] [answered]
Do we cache the APIs in mockoon? What if we want to save this APIs and auto play it again without hitting the main backend-server?
Do we save data (logs seen on GUI)?

Please comment. Thank you.

**Comment by 255kb** [comment, marked as answer]
What do you mean by caching? If you mean being able to start the servers without the UI this is not yet possible. But when using the CLI without the `--daemon-off` flag, the processes are managed by [PM2](https://pm2.keymetrics.io/) which could help you to manage the processes 
Concerning the logs, if you are using the CLI, they are saved in `~/.mockoon-cli/logs/`. They are not saved when using the UI.

**Reply by ankita-sharma2019** [reply]
Hi @255kb , 

Thanks for the response.

This is what I am expected from Mockoon - I want an environment that is the proxy for my actual backend server. It stores/saves the data (request/response). Next time when I hit the mockoon with a request... it should return the stored response for now- then if I can get it then next I would want to manipulate the stored response and send that as the response

The way I am thinking mockoon would help me is - 
1. Find out if the server caches (saves) the response data
2. How this stored data can be used as reponse when the mockoon gets a request?
3. How do I manipulate the resonse on the fly when needed and send the response?

Is this requirement achievable with Mockoon?

Thank you

**Reply by 255kb** [reply]
I understand what you want.
Mockoon cannot replay responses like this, or let you intercept and manipulate them. It is mostly stateless.
I think you will have more luck with other mocking tools that can do exactly what you need (maybe wiremock?)

**Reply by ankita-sharma2019** [reply]
Ok, Thanks
One more query - 


I have both these APIs mocked. By why is the first API request caught by the 2nd one? I am getting wrong responses because of this - What should be done to get the expected?

Please help. Thank you

**Reply by 255kb** [reply]
Mockoon is using [Express](https://expressjs.com/en/guide/routing.html), and colons are used to describe route parameters. So, when you write `/xxxx:yyyy` the route will be `xxxx` with a route param `yyyy`. 
If you want to use a colon in your path, you must escape it like this: `xxxx\\:yyyy`. Apparently, `xxxx[:]yyyy` is working too.

**Reply by ankita-sharma2019** [reply]
Do you mean - While Mocking the Route if We have route params use `[]` around the `:`?

So both the saved Routes would be like - 

`xxxx[:]yyyy`
`abcd[:]efgh` and so on 

Then it should be able to give proper, desired responses

**Reply by 255kb** [reply]
What I mean is that you cannot use a colon `:` in a route definition like a normal character. It will be considered as a param.
So, if you want to use a colon, you need to escape it (with either square brackets or double backslashes) so it does not create a route parameter.

**Reply by ankita-sharma2019** [reply]
Got it, Thanks 

FYI - square brackets worked for me but double backslashes did not

**Reply by 255kb** [reply]
ok, thanks for the info, I will retest :)

**Reply by ankita-sharma2019** [reply]
Hi Guillaume,

Is it possible to split the mock env json file? If yes, how?

I have mocked APIs which has huge responses, it will become difficult to manage the file as we keep on adding new routes with their responses. Hence, wanted to know if there is some way to split my mock environment.

**Reply by 255kb** [reply]
Hi
This is not possible.
One solution is to move the body contents in files, so they are not part of the environment file anymore. Files also support the templating system.

**Reply by ankita-sharma2019** [reply]
OK, Thank you
